Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM golang:1.11.10-alpine as builder
+MAINTAINER FullStory Engineering
+
+RUN apk update && apk add --no-cache ca-certificates git gcc g++ libc-dev
+# create non-privileged group and user
+RUN addgroup -S grpcurl && adduser -S grpcurl -G grpcurl
+
+# build grpcurl
+WORKDIR /tmp/fullstorydev/grpcurl
+# copy just the files/sources we need to build grpcurl
+COPY VERSION *.go go.* /tmp/fullstorydev/grpcurl/
+COPY cmd /tmp/fullstorydev/grpcurl/cmd
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -o /grpcurl -ldflags "-w -extldflags \"-static\" -X \"main.version=$(cat VERSION)\"" ./cmd/grpcurl
```

Comment:
I got this from a website that provided tips for getting a binary that would run in a `scratch` container (which doesn't even have `libc`). The relevant tidbits are the linker flags and the `CGO_ENABLED=0`, so it doesn't try to use any other shared libraries for parts of the standard runtime.

Reply:
SGTM.

nit: readability could probably be improved by either adding those flag as env vars or using `\` to break up the lines
```
ENV CGO_ENABLED=0
ENV GOOS=linux
ENV GOARCH=amd64
ENV GO111MODULE=on 
RUN go build -o /grpcurl -ldflags "-w -extldflags \"-static\" -X \"main.version=$(cat VERSION)\"" ./cmd/grpcurl
```

```
RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on \
    go build -o /grpcurl -ldflags "-w -extldflags \"-static\" -X \"main.version=$(cat VERSION)\"" ./cmd/grpcurl
```